src/crewai/crew.py

```diff
@@ -331,7 +331,7 @@ def create_crew_knowledge(self) -> "Crew":
                     self.knowledge = Knowledge(
                         sources=self.knowledge_sources,
                         embedder=self.embedder,
-                        collection_name="crew",
+                        collection_name=self.name,
```

Comment:
Would you mind to adding a test for that?

Reply:
good call, sure, I will